### DISCUSSION TITLE
Image identification.

### DISCUSSION THREAD

**Discussion by mohammadreza-plutoflume** [dlemstra/Magick.NET#1503] [Help] [answered]
Hi,
Sorry if this is a duplicate question, I couldn't find the answer anywhere!

I have a `byte[]` and I want to find if the data is an image or not, if that is an image what's the format? `png`, `bmp`, `jpg` or not identifiable at all!

Currently, I wrote the following code:

```
using (var image = new MagickImage(fileBytes))
{
    // Converting to PNG
    image.Write(pngMemoryStream, MagickFormat.Png);

    var imageBytes = pngMemoryStream.ToArray();
    // Writing to a png file
}
```

But the `MagickImage` throws an exception when the file is not image, it's legitimate to throw but is there a way I can identify before calling the constructor to prevent throwing? (I don't wanna use try/catch)

Also, I saw I have some `svg` files and the constructor also throws for those svg files! If I could identify first, I could pass `MagickFormat.Svg` to the constructor!

**Comment by dlemstra** [comment, marked as answer]
You can use `MagickFormatInfo.Create(fileBytes)` for this. It will return `null` when the format could not be determined.

**Reply by mohammadreza-plutoflume** [reply]
Great, thank you so much
